app/services/audit/AuditService.scala

```diff
@@ -26,15 +26,39 @@ import uk.gov.hmrc.play.audit.AuditExtensions
 import uk.gov.hmrc.play.audit.http.connector.AuditResult.{Disabled, Failure, Success}
 import uk.gov.hmrc.play.audit.http.connector.{AuditConnector, AuditResult}
 import uk.gov.hmrc.play.audit.model.{DataEvent, ExtendedDataEvent}
+import uk.gov.hmrc.wco.dec.inventorylinking.movement.request.InventoryLinkingMovementRequest
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class AuditService @Inject()(connector: AuditConnector, @Named("appName") appName: String)(
   implicit ec: ExecutionContext
 ) {
-
   private val logger = Logger(this.getClass)
 
+  def auditShutMucrData(eori: String, mucr: String, result: String): Map[String, String] =
```

Comment:
Seems a bit odd calling these methods to create the data then having to call "audit" with the data and choice.  Why not have these "data builder" methods call "audit" with the relevant choice

Reply:
Then maybe "audit" can be private (if it's not called/used elsewhere)?

Reply:
That is a very good point. It seems that everyone got that same conclusion.
The `auditShutMucrData` `auditDisassociateData` were in my first iteration part of the controller/service layers. I then refactored them to put all inside the `AuditService`. :)
Excellent advice.

Reply:
This has been addressed.